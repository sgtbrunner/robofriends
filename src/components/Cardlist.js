import React from 'react';
import Card from './Card';

const Cardlist = ({ robots }) => {
  return (
    <div>
      {
			    robots.map((user) => {
			      return (
  <Card
    name={user.name}
    email={user.email}
    key={user.id}
  />
			      );
			    })
			}
    </div>
  );
};

export default Cardlist;
