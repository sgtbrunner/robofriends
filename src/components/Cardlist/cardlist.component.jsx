import React from 'react';
import PropTypes from 'prop-types';

import Card from '../Card';

const Cardlist = ({ robots }) => (
  <div>
    {robots.map((user) => {
      const { name, email, id } = user;
      return <Card name={name} email={email} id={id} key={id} />;
    })}
  </div>
);

Cardlist.propTypes = {
  robots: PropTypes.array.isRequired,
};

export default Cardlist;
