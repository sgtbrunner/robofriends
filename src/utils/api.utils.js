export const USERS_URL = 'https://jsonplaceholder.typicode.com/users';

export default {
  getUsers: () =>
    fetch(USERS_URL).then((response) => {
      if (response.ok) return response.json();
      throw new Error('Something went wrong');
    }),
};
