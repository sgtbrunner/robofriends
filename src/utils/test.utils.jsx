import React from 'react';
import { render } from '@testing-library/react';

export const mountComponent = ({ component, props = {} }) => {
  const Component = component;
  return render(<Component {...props} />);
};

export default null;
