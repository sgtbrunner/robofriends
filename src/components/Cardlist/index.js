import Cardlist from './cardlist.component';

export default Cardlist;
