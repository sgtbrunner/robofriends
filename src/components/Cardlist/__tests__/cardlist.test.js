import { screen } from '@testing-library/react';

import { mountComponent } from '../../../utils/test.utils';
import USERS from '../../../mocks/users';
import Cardlist from '../index';

const props = {
  robots: USERS,
};

describe('Cardlist component', () => {
  it('should match snapshot', () => {
    const { container } = mountComponent({ component: Cardlist, props });
    expect(container).toMatchSnapshot();
  });

  it('should render five (5) cards if robots has 5 valid items', () => {
    mountComponent({ component: Cardlist, props });
    const cards = screen.getAllByRole('img', { name: 'robot' });
    expect(cards).toHaveLength(5);
  });

  it('should render no (0) cards if robots has no valid items', () => {
    mountComponent({ component: Cardlist, props: { robots: [] } });
    const cards = screen.queryByRole('img', { name: 'robot' });
    expect(cards).not.toBeInTheDocument();
  });
});
