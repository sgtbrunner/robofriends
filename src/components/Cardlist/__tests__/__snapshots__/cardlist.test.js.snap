// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cardlist component should match snapshot 1`] = `
<div>
  <div>
    <div
      class="dib br3 pa1 ma2 grow bw2 shadow-5 bg-light-green"
    >
      <img
        alt="robot"
        src="https://robohash.org/Baby Yodababy.yoda@newrepublic.com1"
      />
      <h2>
        Baby Yoda
      </h2>
      <p>
        baby.yoda@newrepublic.com
      </p>
    </div>
    <div
      class="dib br3 pa1 ma2 grow bw2 shadow-5 bg-light-green"
    >
      <img
        alt="robot"
        src="https://robohash.org/Pep Guardiolapep.guardiola@mancity.co.uk2"
      />
      <h2>
        Pep Guardiola
      </h2>
      <p>
        pep.guardiola@mancity.co.uk
      </p>
    </div>
    <div
      class="dib br3 pa1 ma2 grow bw2 shadow-5 bg-light-green"
    >
      <img
        alt="robot"
        src="https://robohash.org/Supermansuper.man@dccomics.com3"
      />
      <h2>
        Superman
      </h2>
      <p>
        super.man@dccomics.com
      </p>
    </div>
    <div
      class="dib br3 pa1 ma2 grow bw2 shadow-5 bg-light-green"
    >
      <img
        alt="robot"
        src="https://robohash.org/Tim Ferristim.ferris@fourhourworkweek.net4"
      />
      <h2>
        Tim Ferris
      </h2>
      <p>
        tim.ferris@fourhourworkweek.net
      </p>
    </div>
    <div
      class="dib br3 pa1 ma2 grow bw2 shadow-5 bg-light-green"
    >
      <img
        alt="robot"
        src="https://robohash.org/Alexander Supertrampchris.mccandless@intothewild.com5"
      />
      <h2>
        Alexander Supertramp
      </h2>
      <p>
        chris.mccandless@intothewild.com
      </p>
    </div>
  </div>
</div>
`;
